### PR TITLE
Add comprehensive tests for various components and hooks

### PR DESCRIPTION
**Pull request by jsaarimaa**
- Enhance NestedList tests to cover loading states, error handling, and data rendering.
- Extend Preview tests to handle cases where visualizationResponse is null or undefined.
- Improve SaveDialog tests to verify saving behavior based on selected options and publication webhook state.
- Introduce tests for useSelections hook to validate selection management.
- Add assertion tests for TabPanel to ensure correct rendering and behavior based on selected values.
- Implement tests for DimensionList to check rendering and interaction with dimensions.
- Expand navigationContext tests to handle multiple sequential state updates.
- Add edge case tests for useHierarchyParams and useQueryParams hooks.
- Enhance useReplaceQueryParams tests to ensure navigation behavior based on context state.
- Introduce tests for useScrollToElement to validate scrolling behavior and default offsets.
- Add tests for ApiHelpers to validate default query options and language string parsing.
- Extend ChartSettingHelpers tests to cover additional visualization types and edge cases.
- Improve componentHelpers tests by removing unnecessary truthy checks.
- Add tests for dimensionSelectionHelpers to validate default filters and query type labels.
- Enhance editorHelpers tests to cover edge cases for dimension resolution and error handling.
- Add tests for metadataUtils to handle null and undefined additional properties.
- Extend sortingHelpers tests to cover edge cases for sorting dimensions and database items.
- Improve Editor tests to validate loading states, error handling, and dimension display.
- Enhance QueryLoader tests to cover error scenarios and loading states.

### CODE REVIEW

File: PxGraf.Frontend/src/utils/sortingHelpers.test.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ const mockDimensions: IDimension[] =
         code: "Vuosi",
         name: {
             fi: "Vuosi",
-            sv: "�r",
+            sv: "�r",
```
Comment (sakari-malkki) [inline PxGraf.Frontend/src/utils/sortingHelpers.test.ts]: Encoding issue with umlauts in this file?

Comment (jsaarimaa) [reply PxGraf.Frontend/src/utils/sortingHelpers.test.ts]: Fixed

File: PxGraf.Frontend/src/api/services/visualization.test.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        mockPostAsync.mockResolvedValueOnce(mockData);
+
+        const { result } = renderHook(
+            () => useVisualizationQuery(mockIdStack, mockQuery, mockCubeQuery, 'fi', 'lineChart', settings),
+            { wrapper: createWrapper() }
+        );
+
+        await waitFor(() => expect(result.current.isLoading).toBe(false));
+        expect(mockPostAsync).toHaveBeenCalledWith('creation/visualization', expect.any(String));
+    });
```
[the remark this answers is not in the document]
Comment (jsaarimaa) [reply PxGraf.Frontend/src/api/services/visualization.test.ts:106]: Fixed the issue.

**Review by sakari-malkki — APPROVED** [review]
